packages/web-client/app/utils/image.ts

```diff
@@ -89,11 +91,59 @@ export class ImageValidation {
       imageSizeValid = await this.imageSizeWithinBounds(file);
     }
 
+    let message = '';
+
+    if (!fileTypeValid) {
+      const { fileType } = this;
+      const subtypeOnly = fileType.map((v) => v.replace(/^image\//, ''));
+      if (subtypeOnly.length === 1) {
+        message = `Please upload an image with a file type of ${subtypeOnly[0]}`;
+      } else {
+        const lastItem = subtypeOnly.pop();
+        message = `Please upload an image with a file type of ${subtypeOnly.join(
+          ', '
+        )} or ${lastItem}`;
+      }
+    } else if (!fileSizeValid) {
+      const { minFileSize, maxFileSize } = this;
+      if (maxFileSize === Infinity) {
+        message = `Please upload a file with size greater than ${fileSize(
+          minFileSize
+        )}`;
+      } else if (minFileSize <= 0) {
+        message = `Please upload a file with size less than ${fileSize(
+          maxFileSize
+        )}`;
+      } else {
+        message = `Please upload a file between ${fileSize(
+          minFileSize
+        )} and ${fileSize(maxFileSize)}`;
+      }
+    } else if (!imageSizeValid) {
+      const { minWidth, maxWidth, minHeight, maxHeight } = this;
+      // ignoring the case where we want to restrict width but not height
+      if (
+        maxWidth !== Infinity &&
+        maxHeight !== Infinity &&
+        minWidth > 0 &&
+        minHeight > 0
+      ) {
+        message = `Please upload an image larger than ${minWidth}x${minHeight}, and smaller than ${maxWidth}x${maxHeight}`;
+      } else if (minWidth > 0 && minHeight > 0) {
+        message = `Please upload an image larger than ${minWidth}x${minHeight}`;
+      } else if (maxWidth !== Infinity && maxHeight !== Infinity) {
+        message = `Please upload an image smaller than ${maxWidth}x${maxHeight}`;
+      } else {
+        message = `Please upload an image larger than ${minWidth}x${minHeight}`;
```

Comment:
maybe it doesn’t matter but isn’t it technically “at least” vs “larger than”? (same for file size)

I wonder whether the `boolean` nature of `imageSizeWithinBounds` could be decomposed so the calculation doesn’t need to happen twice-ish. But probs not important